### ISSUE TITLE
HOW CAN WE ADD SCREEN SHARING OPTION ??

### ISSUE THREAD

**Issue by PRANEETH-REDDY-23899** [CleverProgrammers/nodejs-zoom-clone#7]
can you please let me knew how can i add a screen sharing option to it

**Comment by idsus** [comment]
I need same help I will support this question so people see it.

**Comment by Pratik-Hub** [comment]
Guys I am creating my own Video Chat App but not using this I have used react and etc... If you want to include share screen in this wonder project of Zoom-Clone then You may use my API and do it!. I will release it in few weeks!!!!

Is That Cool!!! 😎😎😎

**Comment by micaiah-effiong** [comment]
Also have it in mind the mobile browsers don't support screen sharing

**Comment by Pratik-Hub** [comment]
> Also have it in mind the mobile browsers don't support screen sharing


It does works Man

Maybe I am wrong. If yes please forgive me. I am testing my API and it's working correctly in  all devices including screenshare in mobile

**Comment by ChandanRabha** [comment]
Yes please i have been trying to find sources where screen sharing is possible. I even tried changing the mediastream of my webcam video stream to screen share video stream but it then the screen on peer side just freezes as soon as i changed the video stream. Would really be big help if someone finds a way. Thank You

**Comment by pankajjossy1** [comment]
> Guys I am creating my own Video Chat App but not using this I have used react and etc... If you want to include share screen in this wonder project of Zoom-Clone then You may use my API and do it!. I will release it in few weeks!!!!
> 
> Is That Cool!!! 😎😎😎

how pls tell how can i add screen share

**Comment by ChitrangSharma** [comment]
For all of you guys, who want to integrate Screen-Sharing option in the project .
you can get what you want to build here in this channel link below..
**https://youtu.be/X8QHHB7DA90**
I hope it will help you

**Comment by ChitrangSharma** [comment]
> > Guys I am creating my own Video Chat App but not using this I have used react and etc... If you want to include share screen in this wonder project of Zoom-Clone then You may use my API and do it!. I will release it in few weeks!!!!
> > Is That Cool!!! 😎😎😎
> 
> how pls tell how can i add screen share

well if you want to make it, super quick I have given a link there , okay let me paste the link here as well https://youtu.be/X8QHHB7DA90 
in this video you will get how to make screen sharing possible in this application .Thanks.

**Comment by Boopalan-M** [comment]
> can you please let me knew how can i add a screen sharing option to it

i also having that question.Are u acheived that one

**Comment by ttejas123** [comment]
i added screenShare via replacing video stream with screen share.
and user who started screen share == can't see screen share
       other user connected   == can see screen share;
for that I use this code down below

//IN HTML
//add button
<button onClick= "screenshare()"></button>

//IN JS FILE
//screenShare
const screenshare = () =>{
 navigator.mediaDevices.getDisplayMedia({ 
     video:{
       cursor:'always'
     },
     audio:{
            echoCancellation:true,
            noiseSupprission:true
     }

 }).then(stream =>{
     let videoTrack = stream.getVideoTracks()[0];
         videoTrack.onended = function(){
           stopScreenShare();
         }
         for (let x=0;x<currentPeer.length;x++){
           
           let sender = currentPeer[x].getSenders().find(function(s){  //replace video track 
              return s.track.kind == videoTrack.kind;
            })
            
            sender.replaceTrack(videoTrack);
       }
   
  })
  
 }

function stopScreenShare(){
  let videoTrack = myVideoStream.getVideoTracks()[0];
  for (let x=0;x<currentPeer.length;x++){
          let sender = currentPeer[x].getSenders().find(function(s){
              return s.track.kind == videoTrack.kind;
            }) 
          sender.replaceTrack(videoTrack);
  }       
}

**Comment by ttejas123** [comment]
I added screenShare via replacing video stream with screen share.
and user who started screen share == can't see screen share
other user connected == can see screen share;
for that I use this code down below

`//IN HTML
//add button
//IN JS FILE
//screenShare
const screenshare = () =>{
navigator.mediaDevices.getDisplayMedia({
video:{
cursor:'always'
},
audio:{
echoCancellation:true,
noiseSupprission:true
}

}).then(stream =>{
let videoTrack = stream.getVideoTracks()[0];
videoTrack.onended = function(){
stopScreenShare();
}
for (let x=0;x<currentPeer.length;x++){

       let sender = currentPeer[x].getSenders().find(function(s){  //replace video track 
          return s.track.kind == videoTrack.kind;
        })
        
        sender.replaceTrack(videoTrack);
   }
})

}

function stopScreenShare(){
let videoTrack = myVideoStream.getVideoTracks()[0];
for (let x=0;x<currentPeer.length;x++){
let sender = currentPeer[x].getSenders().find(function(s){
return s.track.kind == videoTrack.kind;
})
sender.replaceTrack(videoTrack);
}
}`

**Comment by Boopalan-M** [comment]
> I added screenShare via replacing video stream with screen share.
> and user who started screen share == can't see screen share
> other user connected == can see screen share;
> for that I use this code down below
> 
> `//IN HTML
> //add button
> //IN JS FILE
> //screenShare
> const screenshare = () =>{
> navigator.mediaDevices.getDisplayMedia({
> video:{
> cursor:'always'
> },
> audio:{
> echoCancellation:true,
> noiseSupprission:true
> }
> 
> }).then(stream =>{
> let videoTrack = stream.getVideoTracks()[0];
> videoTrack.onended = function(){
> stopScreenShare();
> }
> for (let x=0;x<currentPeer.length;x++){
> 
> ```
>    let sender = currentPeer[x].getSenders().find(function(s){  //replace video track 
>       return s.track.kind == videoTrack.kind;
>     })
>     
>     sender.replaceTrack(videoTrack);
> ```
> 
> }
> })
> 
> }
> 
> function stopScreenShare(){
> let videoTrack = myVideoStream.getVideoTracks()[0];
> for (let x=0;x<currentPeer.length;x++){
> let sender = currentPeer[x].getSenders().find(function(s){
> return s.track.kind == videoTrack.kind;
> })
> sender.replaceTrack(videoTrack);
> }
> }`

i received error as length is not defined.

**Comment by Boopalan-M** [comment]
> I added screenShare via replacing video stream with screen share.
> and user who started screen share == can't see screen share
> other user connected == can see screen share;
> for that I use this code down below
> 
> `//IN HTML
> //add button
> //IN JS FILE
> //screenShare
> const screenshare = () =>{
> navigator.mediaDevices.getDisplayMedia({
> video:{
> cursor:'always'
> },
> audio:{
> echoCancellation:true,
> noiseSupprission:true
> }
> 
> }).then(stream =>{
> let videoTrack = stream.getVideoTracks()[0];
> videoTrack.onended = function(){
> stopScreenShare();
> }
> for (let x=0;x<currentPeer.length;x++){
> 
> ```
>    let sender = currentPeer[x].getSenders().find(function(s){  //replace video track 
>       return s.track.kind == videoTrack.kind;
>     })
>     
>     sender.replaceTrack(videoTrack);
> ```
> 
> }
> })
> 
> }
> 
> function stopScreenShare(){
> let videoTrack = myVideoStream.getVideoTracks()[0];
> for (let x=0;x<currentPeer.length;x++){
> let sender = currentPeer[x].getSenders().find(function(s){
> return s.track.kind == videoTrack.kind;
> })
> sender.replaceTrack(videoTrack);
> }
> }`

i use this code i didn't get any error. but the screen is not visible to others..

**Comment by idsus** [comment]
Thanks!

**Comment by anujmittal2308** [comment]
https://stackoverflow.com/questions/72003682/cant-fix-expresspeerserver-is-not-a-function-in-server-js
